source/lac/petsc_communication_pattern.cc

```diff
@@ -54,7 +54,9 @@ namespace PETScWrappers
 
     PetscLayout layout;
     AssertPETSc(PetscLayoutCreate(communicator, &layout));
-    AssertPETSc(PetscLayoutSetLocalSize(layout, local_size));
+    const auto petsc_local_size = static_cast<PetscInt>(local_size);
```

Comment:
Why not just
```suggestion
    const PetscInt petsc_local_size = local_size;
```
? Do you want to emphasize that we are casting here?

Reply:
I think we need `static_cast()` to avoid compiler warnings about narrowing, so yes - we should be explicit in the casts.